### PR TITLE
post-checks: add production smoke test

### PR DESCRIPTION
**Pull request by jonalvarezz**
add a test script to ensure parachain node and enclave worker are healthy. 

the script currently only targets the production environment (tee-dev)

it ensures:
1. clients can connect to the Parachain's Node and the Enclave's Worker over WSS.
2. a registered Enclave worker exists in the Parachain's storage.
3. the registered Enclave matches the running one.
4. worker's state is readable and consistent

we plan to run this script automatically after deployment.

reviewers: the meaty stuff lives in `tee-worker/ts-tests/post-checks/tests/post-check.test.ts`
 
**TODO:**
- [ ] `post-checks.yml`: remove the "pull_request" running trigger before merging (used for testing only). and keep it as workflow_dispatch only

### CODE REVIEW

File: tee-worker/ts-tests/post-checks/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "description": "",
+  "type": "module",
+  "main": "index.js",
+  "scripts": {
+    "start": "mocha --timeout 20000 --exit --sort -r ts-node/register --loader=ts-node/esm ./tests/**/*.test.ts"
+  },
+  "dependencies": {
+    "@litentry/enclave": "^4.0.0",
+    "@litentry/parachain-api": "file:../../client-api/parachain-api",
+    "@litentry/sidechain-api": "file:../../client-api/sidechain-api",
```
Comment (0xverin) [inline tee-worker/ts-tests/post-checks/package.json]: I see that you have not added the generation steps for client-api in your yml file, so the deps here cannot be installed, I guess. For reference: https://github.com/litentry/litentry-parachain/blob/dev/tee-worker/cli/lit_ts_integration_test.sh#L57-L68

Comment (jonalvarezz) [reply tee-worker/ts-tests/post-checks/package.json]: right! I forgot we don't track the generated files. I may just install it from NPM 🤔 
thank you, Verin!

File: .github/workflows/post-checks.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Parachain and Enclave integrety checks
+
+on:
+  pull_request:
```
Comment (Kailai-Wang) [inline .github/workflows/post-checks.yml]: Is it supposed to be run on PR?

File: .github/workflows/post-checks.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment"
+        type: choice
+        options:
+          - litentry-prod
+          - litentry-dev
```
Comment (Kailai-Wang) [inline .github/workflows/post-checks.yml]: I'm wondering why there is no `staging` 😂

**Review by Kailai-Wang — COMMENTED** [review]
I'm fine with the tests, and I'm wondering:
- will the test work for `dev` where everything is reset upon each deployment?
- when shall we trigger it? Manually after each deployment? Can this be integrated with Jenkins?

This _could_ be integrated with the tooling website too - like a tool box for the identity-worker setup : )

File: .github/workflows/post-checks.yml (hunk 2 above)
Comment (jonalvarezz) [reply .github/workflows/post-checks.yml]: Nah, I added it just to ensure the github action was well set up 😄 
I left a ToDo in the description to remind me to remove this before merging in

File: .github/workflows/post-checks.yml (hunk 3 above)
Comment (jonalvarezz) [reply .github/workflows/post-checks.yml]: right, missed it!

**Comment by jonalvarezz** [comment]
> * will the test work for `dev` where everything is reset upon each deployment?

@Kailai-Wang, not really, i was actually just targeting the production environment. i should remove the input options from the workflow_dispatch not to cause confusion.

however, this can be a start to having end-to-end tests to run after deploy. I'm happy to take on that task


> * when shall we trigger it? Manually after each deployment? Can this be integrated with Jenkins?


I used a github action so we can trigger it manually at anytime. I'll ask Simple to hook this at the end of his scripts to run it automatically after each prod deploy.

**Review by Kailai-Wang — APPROVED** [review]
Thanks!

**Comment by m1iktea** [comment]
> > * will the test work for `dev` where everything is reset upon each deployment?
> 
> @Kailai-Wang, not really, i was actually just targeting the production environment. i should remove the input options from the workflow_dispatch not to cause confusion.
> 
> however, this can be a start to having end-to-end tests to run after deploy. I'm happy to take on that task
> 
> > * when shall we trigger it? Manually after each deployment? Can this be integrated with Jenkins?
> 
> I used a github action so we can trigger it manually at anytime. I'll ask Simple to hook this at the end of his scripts to run it automatically after each prod deploy.

If an automatic check is needed after prod deploy, my recommendation is to integrate it into Jenkins.

**Pull request merged**